packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts

```diff
@@ -324,6 +323,45 @@ describe('core/strategies/verifyPasswordResetCode', () => {
   });
 });
 
+describe('core/strategies/email_and_password/createUserWithEmailAndPassword', () => {
+  let auth: Auth;
+  const serverUser: APIUserInfo = {
+    localId: 'local-id'
+  };
+
+  beforeEach(async () => {
+    auth = await testAuth();
+    mockFetch.setUp();
+    mockEndpoint(Endpoint.SIGN_UP, {
+      idToken: 'id-token',
+      refreshToken: 'refresh-token',
+      expiresIn: '1234',
+      localId: serverUser.localId!
+    });
+    mockEndpoint(Endpoint.GET_ACCOUNT_INFO, {
+      users: [serverUser]
+    });
+  });
+  afterEach(mockFetch.tearDown);
+
+  it('should sign in the user', async () => {
+    const {
+      credential,
+      user,
+      operationType
+    } = await createUserWithEmailAndPassword(
+      auth,
+      'some-email',
+      'some-password'
+    );
+    expect(credential?.providerId).to.eq(ProviderId.PASSWORD);
```

Comment:
nit: Please use `credential!.providerId` if you are asserting that `credential` is not null and has a field. In this way, when `credential` is null you'll get a proper runtime error message saying "cannot read property `providerId` of null", instead of "expected null to equal some-password". The latter doesn't differentiate whether `credential` is null or `credential.providerId` is.

Reply:
Ditto for similar assertions below.

Reply:
done